### ISSUE TITLE
`.env` Hot Reloading Error (Astro v5 Beta)

### ISSUE THREAD

**Issue by neelsudhakaran** [withastro/astro#12107]
### Astro Info

```block
Astro                    v5.0.0-beta.3
Node                     v20.17.0
System                   macOS (arm64)
Package Manager          pnpm
Output                   static
Adapter                  none
Integrations             none
```

### Describe the Bug

When modifying a `.env` file while the dev server is running, a Vite error is logged to the console.

```
[ERROR] [vite] Error: cannot print server URLs before server.listen is called.
```

Sometimes (I can't tell if it's when you make multiple saves in rapid succession or something else), you also get the following Node error thrown.

```
(node:84966) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 unhandledRejection listeners added to [process]. MaxListeners is 10. Use emitter.setMaxListeners() to increase limit
(Use `node --trace-warnings ...` to show where the warning was created)
```

As far as I can tell (and I'm still using environment variables the old way) the environment variables, both with the `PUBLIC_` prefix and without, still work.

### What's the expected result?

I don't think these warnings, if true, are supposed to print.

### Link to Minimal Reproducible Example

https://github.com/neelsudhakaran/astro-server-listen-error

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by ematipico** [comment]
@neelsudhakaran your reproduction doesn't have an `.env` file, can you add it?

**Comment by neelsudhakaran** [comment]
Sorry about that @ematipico! I added the `.env` to the repository. Any edits to the file (or even just saving it) brings up that error for me.

**Comment by ascorbic** [comment]
Fixed in https://github.com/withastro/astro/pull/12127

**Comment by neelsudhakaran** [comment]
It was really interesting to read about the fix—thank you so much by the way!

**Comment by dushmanta05** [comment]
Is this issue fixed in 4.xx? Could you please tell? I got this issue in my Astro 4.16 version project?

**Comment by ascorbic** [comment]
@dushmanta05 It was fixed in 5.0, so you'll need to upgrade

**Comment by dushmanta05** [comment]
@ascorbic Thank you for info.
